### PR TITLE
Clean up the gradle build logic used to generate the Godot Android binaries

### PR DESCRIPTION
**Pull request by m4gr3d**

### CODE REVIEW

**Review by raulsntos — APPROVED** [review]
Can't speak for the Android team, but this looks reasonable to me and helps me implement https://github.com/godotengine/godot/pull/88803. Thank you!

File: platform/android/java/build.gradle [hunk 1]
```diff
@@ -206,79 +107,95 @@ def generateBuildTasks(String flavor = "template") {
...
+                // Copy the Godot android library archive file into the app module libs directory.
+                // Depends on the library build task to ensure the AAR file is generated prior to copying.
+                String copyAARTaskName = "copy${capitalizedTarget}AARToAppModule"
+                if (tasks.findByName(copyAARTaskName) != null) {
+                    buildTasks += tasks.getByName(copyAARTaskName)
+                } else {
+                    buildTasks += tasks.create(name: copyAARTaskName, type: Copy) {
+                        dependsOn ":lib:assembleTemplate${capitalizedTarget}"
+                        from('lib/build/outputs/aar')
+                        include("godot-lib.template_${target}.aar")
```
Comment (raulsntos) [inline platform/android/java/build.gradle:127]: This won't copy the `dev` builds that have a name like: `godot-lib.template_debug.dev.aar`.

Could do something like:

```gradle
String targetSuffix = target
if (target == "dev") {
    targetSuffix = "debug.dev"
}
```

File: platform/android/java/build.gradle [hunk 2]
```diff
@@ -206,79 +107,95 @@ def generateBuildTasks(String flavor = "template") {
...
+                // Copy the Godot android library archive file into the root bin directory.
+                // Depends on the library build task to ensure the AAR file is generated prior to copying.
+                String copyAARToBinTaskName = "copy${capitalizedTarget}AARToBin"
+                if (tasks.findByName(copyAARToBinTaskName) != null) {
+                    buildTasks += tasks.getByName(copyAARToBinTaskName)
+                } else {
+                    buildTasks += tasks.create(name: copyAARToBinTaskName, type: Copy) {
+                        dependsOn ":lib:assembleTemplate${capitalizedTarget}"
+                        from('lib/build/outputs/aar')
+                        include("godot-lib.template_${target}.aar")
```
Comment (raulsntos) [inline platform/android/java/build.gradle:141]: Same here.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**